python-avd/pyavd/_schema/models/avd_model.py

```diff
@@ -457,3 +458,66 @@ def _compare(self, other: object, ignore_fields: tuple[str, ...] = ()) -> bool:
 
         # Use regular __eq__ check on all nested models, since we do not carry over ignore_fields.
         return all(value == other.__dict__[field] for field, value in self.items() if field not in ignore_fields)
+
+    def _combine(self, other: Self) -> None:
+        """
+        Update instance by combining the other instance in.
+
+        Combining is different from merging in the sense that it will raise if there is a conflict
+        between one of our elements and the other elements.
+
+        for AvdIndexedList this is simply an append, if any conflict occurs a duplicate error will be raised.
+
+        Args:
+            other: The other instance of the same type to combine into this instance.
+
+        Raises:
+            AristaAvdDuplicateDataError: If any item from other is conflicting with an item from self when appending.
+        """
+        cls = type(self)
+        if not isinstance(other, cls):
+            msg = f"Unable to combine type '{type(other)}' into '{cls}'"
+            raise TypeError(msg)
+
+        if other._created_from_null:
+            # Force all fields on this instance back to unset if other is a "null" class.
+            self.__dict__ = {}
+            self._custom_data = {}
+
+        for field, new_value in other.items():
+            old_value = self._get_defined_attr(field)
+
+            if old_value is Undefined:
+                # value not set so we can just pick the old_value
+                setattr(self, field, new_value)
+                continue
+
+            # Combine new value
+            field_type = self._fields[field]["type"]
+            if issubclass(field_type, AvdBase):
+                # Merge in to the existing object
+                old_value = cast("AvdBase", old_value)
+                new_value = cast("AvdBase", new_value)
+                old_value._combine(new_value)
+                continue
+
+            if field_type is dict:
+                # TODO: In place combining without schema - need to set the keys probably
```

Comment:
This only happens for a few places with ambigous structured config schemas. For now I think we can just do a comparison like any other "leaf" key.

Reply:
remove the check